query.go

```diff
@@ -0,0 +1,75 @@
+package influxdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+// QueryCSV returns the result of a flux query.
+// TODO: annotations optionally
+func (c *Client) QueryCSV(ctx context.Context, flux string, org string) (*QueryCSVResult, error) {
+	qURL, err := c.makeQueryURL(org)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest("POST", qURL, bytes.NewBufferString(flux))
+	if err != nil {
+		return nil, err
+	}
+	req = req.WithContext(ctx)
+	req.Header.Set("User-Agent", c.userAgent)
+	req.Header.Set("Authorization", c.authorization)
+	req.Header.Set("Content-Type", "application/vnd.flux")
+	resp, err := c.httpClient.Do(req)
+	// this is so we can unset the defer later if we don't error.
+	cleanup := func() {
+		resp.Body.Close()
+	}
+	defer func() { cleanup() }()
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		r := io.LimitReader(resp.Body, 1<<14) // only support errors that are 16kB long, more than that and something is probably wrong.
```

Comment:
I'm confused why this is needed, what are we preventing against here?

Reply:
You want to flush the tcp buffer before closing the connection in order to reuse the connection, but if it is too full, it is faster to just make a new connection and let this one time out.